isis/src/control/objs/BundleUtilities/BundleObservationSolveSettings.cpp

```diff
@@ -47,38 +47,157 @@ namespace Isis {
-   *
-   * @param bundleSettingsFolder Where this settings XML resides - /work/.../projectRoot/images/import1
-   * @param xmlReader An XML reader that's up to an <bundleSettings/> tag.
-   */
-  BundleObservationSolveSettings::BundleObservationSolveSettings(
-                                                             FileName xmlFile,
-                                                             Project *project,
-                                                             XmlStackedHandlerReader *xmlReader) {
-
+  BundleObservationSolveSettings::BundleObservationSolveSettings(const PvlGroup &scParameterGroup) {
     initialize();
-    QString xmlPath = xmlFile.expanded();
-    QFile qXmlFile(xmlPath);
-    if (!qXmlFile.open(QFile::ReadOnly) ) {
-      throw IException(IException::Io,
-                       QString("Unable to open xml file, [%1],  with read access").arg(xmlPath),
-                       _FILEINFO_);
+
+      // group name must be instrument id
+    m_instrumentId = (QString)scParameterGroup.nameKeyword();
+
+    // If CKDEGREE is not specified, then a default of 2 is used
+    if (scParameterGroup.hasKeyword("CKDEGREE")) {
+      m_ckDegree = (int)(scParameterGroup.findKeyword("CKDEGREE"));
     }
 
-    QXmlInputSource xmlInputSource(&qXmlFile);
-    xmlReader->pushContentHandler(new XmlHandler(this, project));
-    xmlReader->setErrorHandler(new XmlHandler(this, project));
-    bool success = xmlReader->parse(xmlInputSource);
-    if (!success) {
-      throw IException(IException::Unknown,
-                       QString("Failed to parse xml file, [%1]").arg(xmlPath),
-                        _FILEINFO_);
+    // If CKSOLVEDEGREE is not specified, then a default of 2 is used -------jwb----- why ??? why not match camsolve option ???
+    if (scParameterGroup.hasKeyword("CKSOLVEDEGREE")) {
+      m_ckSolveDegree = (int) (scParameterGroup.findKeyword("CKSOLVEDEGREE"));
+    }
+
+    // do we solve for No pointing, ANGLES only, ANGLES+ANGULAR VELOCITY, ANGLES+ANGULAR VELOCITY+
+    // ANGULAR ACCELERATION, or a higher order polynomial
+    QString csolve = "NONE";
+    csolve = (QString)scParameterGroup.findKeyword("CAMSOLVE");
+    csolve = csolve.toUpper();
+    if (csolve == "NONE") {
+      m_instrumentPointingSolveOption = NoPointingFactors;
+      m_numberCamAngleCoefSolved = 0;
+    }
+    else if (csolve == "ANGLES") {
+      m_instrumentPointingSolveOption = AnglesOnly;
+      m_numberCamAngleCoefSolved = 1;
+    }
+    else if (csolve == "VELOCITIES") {
+      m_instrumentPointingSolveOption = AnglesVelocity;
+      m_numberCamAngleCoefSolved = 2;
+    }
+    else if (csolve == "ACCELERATIONS") {
+      m_instrumentPointingSolveOption = AnglesVelocityAcceleration;
+      m_numberCamAngleCoefSolved = 3;
+    }
+    else if (csolve == "ALL"){
+      m_instrumentPointingSolveOption = AllPointingCoefficients;
+      m_numberCamAngleCoefSolved = m_ckSolveDegree + 1;
+    }
+
+    m_anglesAprioriSigma.clear();
```

Comment:
This should be moved down to where it is actually set.

Reply:
is it even necessary to clear them? I'm guessing they are there from when this was implemented as the setFromPvl method, but since it's a constructor now this will be the first time that they are set anywhere

Reply:
that's correct these can just go away then.